pygmt/src/meca.py

```diff
@@ -283,6 +283,10 @@ def meca(
 
         # convert dict to pd.DataFrame so columns can be reordered
         if isinstance(spec, dict):
+            # convert values to ndarray so pandas doesn't complain about "all
+            # scalar values". See issue #2174 for details.
+            for key, value in spec.items():
+                spec[key] = np.atleast_1d(value)
```

Comment:
Use a [dictionary comprehension](https://stackoverflow.com/questions/27518016/apply-function-on-all-values-of-dictionary)? Or do we want keep the for-loop to error nicely when invalid values are entered?
```suggestion
            spec = {key: np.atleast_1d(value) for key, value in spec.items()}
```